test/client/metadataTransfer.test.ts

```diff
@@ -211,51 +235,28 @@ describe('MetadataTransfer', () => {
 
       expect(checkStatus.notCalled).to.be.true;
       expect(result).to.deep.equal({ id: '1' });
+      expect(cancelListenerStub.callCount).to.equal(1);
+      expect(updateListenerStub.callCount).to.equal(0);
```

Comment:
Ensures checkStatus is not called with an immediate cancel.  Also ensures that the cancel listener is called while the update listener is not.